### PR TITLE
Virtual keyboard for Steam and mobile

### PR DESCRIPTION
**Pull request by offalynne**

### CODE REVIEW

File: scripts/input_keyboard_virtual_show/input_keyboard_virtual_show.gml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+///          This covers mobile platforms as well as Xbox and Steam
+/// @param   [virtualKeyboardType]
+
+function input_keyboard_virtual_show(_virtual_keyboard_type = kbv_type_default)
+{
+    static __warned = false;    
+    if (!__warned)
+    {
+        if (INPUT_KEYBOARD_TYPE != "virtual") __input_trace("Warning! Onscreen keyboard is not suitable for use on the current platform");
+        if (INPUT_KEYBOARD_TYPE != "async")   __input_trace("Consider using async dialog for modal text input instead");
```
Comment (JujuAdams) [inline scripts/input_keyboard_virtual_show/input_keyboard_virtual_show.gml]: Should this be `(INPUT_KEYBOARD_TYPE == "async")`?

File: scripts/input_keyboard_virtual_show/input_keyboard_virtual_show.gml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        switch (_virtual_keyboard_type)
+        {
+            case kbv_type_email:   _virtual_keyboard_type = steam_floating_gamepad_text_input_mode_email;       break;
+            case kbv_type_numbers: _virtual_keyboard_type = steam_floating_gamepad_text_input_mode_numeric;     break;
+            default:               _virtual_keyboard_type = steam_floating_gamepad_text_input_mode_single_line; break;
+        }
+            
+        steam_show_floating_gamepad_text_input(_virtual_keyboard_type, 0, 0, 0, 0);
+    }
+    else if (__INPUT_ON_ANDROID || !keyboard_virtual_status())
```
Comment (JujuAdams) [inline scripts/input_keyboard_virtual_show/input_keyboard_virtual_show.gml]: Should this be `(__INPUT_ON_ANDROID && !keyboard_virtual_status())`? Should this contain a check to see if the OSK is suitable i.e. `(INPUT_KEYBOARD_TYPE == "virtual")`?

File: scripts/input_keyboard_virtual_show/input_keyboard_virtual_show.gml (hunk 1 above)
Comment (offalynne) [reply scripts/input_keyboard_virtual_show/input_keyboard_virtual_show.gml]: had this fixed this and then apparently.. unfixed it ? ty for review

File: scripts/input_keyboard_virtual_show/input_keyboard_virtual_show.gml (hunk 2 above)
Comment (offalynne) [reply scripts/input_keyboard_virtual_show/input_keyboard_virtual_show.gml]: > *Should this be `(__INPUT_ON_ANDROID && !keyboard_virtual_status())`*

nope, Android wildly misreports `keyboard_virtual_status()` in many scenarios making it basically-useless there. of the 2 other ways to check if it's up (`.._height()`, and an async event), neither reliably work either ! luckily doing `.._show()` when it's already up on Android is painless. so, we're doing it "the right" way on iOS/tvOS and Xbox GDK, and ignoring status on Android.

> *Should this contain a check to see if the OSK is suitable*

tested thoroughly as cross-compatible (`..show()` function is just a stub where not relevant) so I don't mind leaving it open ended for potential future platforms. this is also not *the best* way on Xbox (generally YYG intends the async stuff for the console OS modals) which gets the warning, but can still show the nonmodal OSK. glad to add that safeguard if you'd prefer, tho.

will add some clarifying comments to cover this..

**Pull request merged**